core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java

```diff
@@ -16,46 +16,43 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-
 package org.apache.iceberg.hadoop;
 
+import com.google.common.base.Preconditions;
+import java.io.IOException;
+import java.util.HashMap;
 import java.util.Map;
 import org.apache.hadoop.conf.Configurable;
 import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
 import org.apache.hadoop.fs.Path;
 import org.apache.iceberg.BaseTable;
 import org.apache.iceberg.PartitionSpec;
 import org.apache.iceberg.Schema;
 import org.apache.iceberg.Table;
 import org.apache.iceberg.TableMetadata;
 import org.apache.iceberg.TableOperations;
-import org.apache.iceberg.Tables;
+import org.apache.iceberg.catalog.Catalog;
+import org.apache.iceberg.catalog.TableIdentifier;
 import org.apache.iceberg.exceptions.AlreadyExistsException;
 import org.apache.iceberg.exceptions.NoSuchTableException;
+import org.apache.iceberg.exceptions.RuntimeIOException;
+
+public class HadoopCatalog implements Catalog, Configurable {
 
-/**
- * Implementation of Iceberg tables that uses the Hadoop FileSystem
- * to store metadata and manifests.
- */
-public class HadoopTables implements Tables, Configurable {
   private Configuration conf;
 
-  public HadoopTables() {
+  public HadoopCatalog() {
```

Comment:
I don't think it makes sense to move the Hadoop implementation to be a Catalog. Identifiers don't handle paths well and there are quite a few changes for not much benefit.

What about just moving Hive to use the Catalog interface? Do we need to move HadoopTables or can we leave it as-is?